File: front-end/src/app/models/appointment-dto-get.ts
import { AppointmentStatus } from "./enums/appointment-status";
import { AppointmentType } from "./enums/appointment-type";


export interface AppointmentDtoGet {
  id: number;
  clientFullName: string;
  doctorFullName: string;
  reason: string;
  appointmentDateTime: string;
  status: AppointmentStatus;
  type: AppointmentType;
  departmentName: string;
}

File: back-end/src/main/java/com/tfg/back/model/dtos/appointment/AppointmentDtoGet.java
package com.tfg.back.model.dtos.appointment;

import com.tfg.back.enums.AppointmentStatus;
import com.tfg.back.enums.AppointmentType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AppointmentDtoGet {

    private Long id;
    private String clientFullName;
    private String doctorFullName;
    private String reason;
    private LocalDateTime appointmentDateTime;
    private AppointmentStatus status;
    private AppointmentType type;
    private String departmentName;
    private String diagnosis;
}
